src/app/dataset-create/dataset-create.component.ts

```diff
@@ -1,9 +1,150 @@
-import { ChangeDetectionStrategy, Component } from "@angular/core";
+import { BaseComponent } from "src/app/common/base.component";
+/* eslint-disable @typescript-eslint/unbound-method */
+import { DatasetKind } from "src/app/api/kamu.graphql.interface";
+import { MaybeNull } from "./../common/app.types";
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    OnInit,
+} from "@angular/core";
+import * as monaco from "monaco-editor";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { AppDatasetCreateService } from "./dataset-create.service";
 
 @Component({
     selector: "app-dataset-create",
     templateUrl: "./dataset-create.component.html",
     styleUrls: ["./dataset-create.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DatasetCreateComponent {}
+export class DatasetCreateComponent extends BaseComponent implements OnInit {
+    private readonly kindMapper: Record<string, DatasetKind> = {
+        root: DatasetKind.Root,
+        derivative: DatasetKind.Derivative,
+    };
+    public readonly initialHint = "# You can edit this file\n";
+    public readonly sqlEditorOptions: monaco.editor.IStandaloneEditorConstructionOptions =
+        {
+            theme: "vs",
+            language: "yaml",
+            renderLineHighlight: "none",
+            minimap: {
+                enabled: false,
+            },
+        };
+    public yamlTemplate = "";
+    public fileName: MaybeNull<string> = "";
+    public showMonacoEditor = false;
+    public errorMessage = "";
+    public owners = ["kamu"];
+    public createDatasetForm: FormGroup = this.fb.group({
+        owner: ["kamu", [Validators.required]],
+        datasetName: [
+            "",
+            [
+                Validators.required,
+                Validators.pattern(
+                    /^([a-zA-Z0-9][a-zA-Z0-9-]*)+(\.[a-zA-Z0-9][a-zA-Z0-9-]*)*$/,
```

Comment:
I'd like to see solid test coverage for this validator.

Reply:
Added new test with validation